### ISSUE TITLE
Can we setup company SSO with Opengrok

### ISSUE THREAD

**Issue by sahoode** [oracle/opengrok#3667]
**Describe the bug**
Is there any possibility to set up SSO with open grok. 
JDK 11

Opengrok version :1.7.13

**To Reproduce**
NA

**Expected behavior**
Opengrok application opening with SSO setting

**Screenshots**

**Additional context**

**Comment by ahornace** [comment]
Yes, you can setup HTTP reverse proxy. For example by using Apache and https://github.com/latchset/mod_auth_mellon

**Comment by ahornace** [comment]
Duplicate of https://github.com/oracle/opengrok/issues/3189

**Comment by vladak** [comment]
In general, the important thing to realize is that OpenGrok by itself does not directly deal with authentication, only authorization. For OpenGrok deployments SSO is basically authentication layer; it is transparent to OpenGrok and the web app can consume there results of the SSO in order to perform authorization decisions or it can do nothing (in which case anyone authenticated can access the service if setup correctly). The authentication layer can be implemented as a reverse proxy server like @ahornace says or possibly as a Tomcat Valve.

Let me know if you want more details about deployment with SAML and Apache with `mod_auth_mellon`.

**Comment by sahoode** [comment]
Hello @vladak please help me with the steps to deployment with SAML and Apache with mod_auth_mellon.
for opengrok
